### DISCUSSION TITLE
Can I build and Intranet and external site and let them share data and media?

### DISCUSSION THREAD

**Discussion by DrewBrasher** [OrchardCMS/OrchardCore#13123] [Q&A] [answered]
I'm about to rebuild an ASP.NET Webforms Intranet in .Net 6 and I'm thinking about building it as an Orchard Core CMS site. The way things are currently setup though is that some things that are created in the intranet need to show up on the external site. For example, Some of our employees teach training classes and they manage the classes on the intranet but once they add a class, people outside of our organization can register for them on our public site. Could I do something like that with both the intranet and the external site using Orchard Core CMS?

I did think about just moving the class management stuff off the intranet and into the admin side of the external site but I think my organization likes the security of having the admin stuff only accessible on our network.

Another thing I wondered about is whether or not it's possible to have an Orchard Core CMS that is both a Full CMS and a Headless or Decoupled. Where the Intranet would be the Full CMS and the external site would be the Headless or decoupled part where I would write my own code to interact with Orchard.

If anyone has any thoughts on any of this I would like to know.

**Comment by hishamco** [comment]
Is there anything needed to rebuild your app? If you want to use .NET Core instead of .NET Framework you can follow the migration steps in Micorosoft doc

About Orchard Core you can use it the way you like, you can create a Razor Pages for your admin stuff and use Orchard Core as source of data, also you can build your website with full CMS.

At the end you can secure your end points with permissions so, the website can access some of the functionality

**Comment by Piedone** [comment, marked as answer]
I think the main question is, how much separation do you want between the two apps. Usually, the point of an intranet site (meaning a true intranet one, where it's only accessible from an internal network) is to separate it from the public internet completely, at least in terms of inbound traffic.

To achieve this, you can e.g. make the intranet site push data, like classes, to the public one, via the RESTful content API of Orchard (or a custom API). You can centralize the class management logic to the public site, and let the intranet one interact with that. It would also be possible the other way around, with the public site invoking the API of the intranet one, but I think it would be a bit riskier on network security (but then you can put the class database on the intranet too, which might be a plus).

If the intranet site would really be an internally used site but otherwise available on the public internet, then it's different.

A single app can be both full CMS and offer the APIs of a headless site, yes (it's just about allowing the RESTful and GraphQL APIs). Decoupled is contrary to full CMS though, since it's a question of what drives the frontend, Orchard (full CMS) or something custom you do in the same ASP.NET Core app (like Razor Pages). (But a decoupled site can also open up the Orchard APIs.)

**Reply by DrewBrasher** [reply]
Thank you, this is helpful. I'm going to give some more thought to how we want to proceed.
